src/infra/marketplace/marketplaceCmdRepo.go

```diff
@@ -54,9 +56,13 @@ func (repo *MarketplaceCmdRepo) installServices(
 			shouldCreatePhpVirtualHost = true
 		}
 
-		_, err := servicesQueryRepo.ReadByName(serviceWithVersion.Name)
-		if err == nil {
-			continue
+		readInstalledServiceDto := dto.ReadInstalledServicesItemsRequest{
+			Name:                 &serviceWithVersion.Name,
+			ShouldIncludeMetrics: false,
+		}
+		_, err := servicesQueryRepo.ReadUniqueInstalledItem(readInstalledServiceDto)
```

Comment:
I really don't get the "Unique". Is your intention to avoid a length check? Do you want to retrieve just one item? Pagination isn't for that? If your intention is to avoid the length check, you're programming implicitly cause I can't understand your line of thought here. I would prefer a length check 50x if that's what you were trying to avoid. When I see a ReadUnique for me has to be super clear what Unique means. Unique isn't one. Unique is something original, different. Must have a clear explanation on the code what that means.

Reply:
I conducted extensive research on language patterns in Go to try and find a solution that doesn’t require removing the "FindOne" I attempted to create here. In the end, I think renaming it to "ReadOne" instead of "ReadUnique" is the best option.

In some situations, repetitive result checks using just the conventional "Read" can make code harder to read, which in turn increases its complexity. The example you mentioned about GORM using `Find()` and requiring a check on `RowsAffected` to ensure all records were retrieved is cumbersome for single-record queries. That’s why such operations are abstracted with methods like [`First()`](https://gorm.io/docs/query.html#Retrieving-a-single-object).

I considered using `Must` as prefix, but it has become a reserved term by convention in Go for functions that panic instead of returning errors. In fact, [you can create such functions yourself to manage the return of other functions](https://stackoverflow.com/a/73584801).

I see no downsides to having this type of abstraction for single-record queries, as it’s already common practice in ORMs, for example.